FIX: revisit preprocessed data QC

celprov opened the pull request:
fix: clearly separate QA from exclusion criteria
fix: clearly layout how to go through QCT, BHT and RSfMRI part of the report separately

celprov committed:
fix: clearly separate QA from exclusion criteria
fix: clearly layout how to go through QCT, BHT and RSfMRI part of the report separately

celprov committed:
fix: clearly separate QA from exclusion criteria
fix: clearly layout how to go through QCT, BHT and RSfMRI part of the report separately

celprov commented:
@oesteban I did not re-read this draft. Feel free to wait until I do a second pass or help me get there; up to you.

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -1,7 +1,8 @@
 Because *fMRIPrep* creates a single anatomical reference for all sessions, we generate such reference first by setting the `--anat-only` flag.
 If that *fMRIPrep* execution finishes successfully, the anatomical processing outcomes will be stored in the output folder.
-We will then run one *fMRIPrep* process for each dataset's session, which is the recommended way for datasets with a large number of sessions (e.g., more than six sessions).
+We will then run one *fMRIPrep* process for each dataset's session, which is the recommended way for datasets with a large number of sessions (e.g., more than 
+six sessions).
```
Why this line break?

```suggestion
We will then run one *fMRIPrep* process for each dataset's session, which is the recommended way for datasets with a large number of sessions (e.g., more than  six sessions).
```

celprov replied:
I'm not sure why I have many seemingly random line break ?!

oesteban commented on docs/processing/preprocessing.md:
````diff
@@ -18,7 +19,8 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
         ```
 
-- [ ] Once the anatomical workflow ran successfully, submit a *job array* with one scanning session each with the `--bids-filter-file` argument selecting the corresponding session, and point the `--fs-subjects-dir` argument to the folder where *FreeSurfer* results were stored.
+- [ ] Once the anatomical workflow ran successfully, submit a *job array* with one scanning session each with the `--bids-filter-file` argument selecting the
+    corresponding session, and point the `--fs-subjects-dir` argument to the folder where *FreeSurfer* results were stored.
````
Our markdown is configured to break lines when you add them, so this will render weirdly (I'm 90% sure of this):
```suggestion
- [ ] Once the anatomical workflow ran successfully, submit a *job array* with one scanning session each with the `--bids-filter-file` argument selecting the corresponding session, and point the `--fs-subjects-dir` argument to the folder where *FreeSurfer* results were stored.
```

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -37,7 +39,8 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
     - [ ] Check the "Errors" section of the visual report.
     - [ ] Check the `log/` folder corresponding to the *fMRIPrep* run, carefully ensuring no errors were missed out on the reports.
-    - [ ] Search for associated keywords in the [issues on *fMRIPrep*'s GitHub repository](https://github.com/nipreps/fmriprep/issues) if the solution remains unclear after the first assessment; it is likely someone else experienced the same problem before you and reported it.
+    - [ ] Search for associated keywords in the [issues on *fMRIPrep*'s GitHub repository](https://github.com/nipreps/fmriprep/issues) if the solution remains 
+        unclear after the first assessment; it is likely someone else experienced the same problem before you and reported it.
```
```suggestion
    - [ ] Search for associated keywords in the [issues on *fMRIPrep*'s GitHub repository](https://github.com/nipreps/fmriprep/issues) if the solution remains unclear after the first assessment; it is likely someone else experienced the same problem before you and reported it.
```

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
+- [ ] Assess the "Anatomical conformation" section and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#anatomical-conformation).
+- [ ] Assess the mosaic showing the calculated brain mask and brain tissue segmentation, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#brain-mask-and-brain-tissue-segmentation-of-the-t1w).
+- [ ] Visualize the spatial normalization flickering mosaic, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#spatial-normalization-of-the-anatomical-t1w-reference).
+    Flickering between the subject and the template space is active while hovering your mouse on the mosaic area.
```
```suggestion
    Flickering between the subject and the template space is active while your mouse pointer hovers the mosaic area.
```

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
+- [ ] Visualize the spatial normalization flickering mosaic, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#spatial-normalization-of-the-anatomical-t1w-reference).
+    Flickering between the subject and the template space is active while hovering your mouse on the mosaic area.
+- [ ] Assess the surface reconstruction mosaic, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#surface-reconstruction).
+- [ ] Visualize the first section entitled `Summary` and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#summary).
```
```suggestion
- [ ] Visualize the first section entitled *Summary* and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#summary).
```

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
     !!! warning "Running through the visual report does not have to be executed in a fixed order"
 
-        While we describe the run through the report as linear, often inspecting other reportlets can help make a decision about the exclusion criteria related to the reportlet at hand. 
+        While we describe the run through the report as linear, often inspecting other reportlets can help make a decision about the exclusion criteria related 
+        to the reportlet at hand. 
```
linebreak
```suggestion
        While we describe the run through the report as linear, often inspecting other reportlets can help make a decision about the exclusion criteria related to the reportlet at hand. 
```

Please update to follow the directions we have sketched in the previous PR to qc of unprocessed data (there's a similar admonition that I have edited for clarity).

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
+    images to evaluate the quality of fMRI scans and derived constructs throughout the whole analysis pipeline. We are however going through the *MRIQC* reports 
+    to train our eye, anticipate issues that might be affecting RSfMRI and flag the corresponding sessions.
+
+- [ ] Go through the report section of the QCT
+    - [ ] Scrutinize the textual summary and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#textual-summary).
+    - [ ] Visualize the T2* map mosaic, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#t2-map).
+    - [ ] Check the T2* gray-matter intensity histogram, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#t2-gray-matter-values).
+    - [ ] Visualize the co-registration flickering mosaic, apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#alignment-of-functional-and-anatomical-mri-data).
+        Flickering between T1w and BOLD images is active while hovering your mouse on the mosaic area.
+    - [ ] Skip the next reportlets and go to the BHT section
```
If this is the same below, please merge and indicate that you sequentially do this on the QCT and then BHT.

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
+- [ ] Visualize the carpet plot and nuisance signals panel, and apply the [QA/QC criteria](qaqc-criteria-preprocessed.md#bold-summary).
+- [ ] Visualize the confound correlation heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-between-nuisance-regressors) 
+    you will include in the nuisance regression model.
+
```
```suggestion
```

oesteban commented on docs/processing/preprocessing.md:
```diff
@@ -49,38 +52,70 @@ We avert that session-wise *fMRIPrep*'s processes run into race conditions by pr
+- [ ] Visualize the confound correlation heatmap and use it to [choose the regressors](qaqc-criteria-preprocessed.md#correlations-between-nuisance-regressors) 
+    you will include in the nuisance regression model.
+
+- [ ] Finally, verify that no errors are reported within the "Errors" section. If there are, proceed as indicated in ["Not all *fMRIPrep* derivatives were generated"](#executing-fmriprep).
```
Please edit like suggested in the qc of unprocessed.
```suggestion
- [ ] Finally, verify that no errors are reported within the "Errors" section.
    If there are, proceed as indicated in ["Not all *fMRIPrep* derivatives were generated"](#executing-fmriprep).
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
 ## Summary
+- [ ] Check that the anatomical processing is based on the number of T1w images you expected and that no excluded T1w was mistakenly included in the processing.
+  If that is not the case, use the *fMRIPrep* `--bids-filter-file` flag to exclude unwanted T1w from the processing and re-run *fMRIPrep*.
+- [ ] Verify that the standard output space corresponds to the one you want.
```
We should say which one is that:
```suggestion
- [ ] Verify that the standard output space corresponds to the one you want.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
     - [ ] Check that the magenta line outlines ventricles.
     - [ ] Check that the contours do not exclude single voxels within piecewise-smooth regions (generally more identifiable in the WM and inside the ventricles).
-      Careful if those single excluded voxels only appear close to tissue boundaries, they are likely misclassified because of partial volume effect, as such it does not consistitute an exclusion criteria.
+      Careful if those single excluded voxels only appear close to tissue boundaries, they are likely misclassified because of partial volume effect, as such it does not constitute an exclusion criteria.
```
```suggestion
      Be careful if those single excluded voxels only appear close to tissue boundaries.
      They are likely misclassified because of the partial volume effect, and as such, they do not constitute an exclusion criteria.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
+      Careful if those single excluded voxels only appear close to tissue boundaries, they are likely misclassified because of partial volume effect, as such it does not constitute an exclusion criteria.
     - [ ] Check that the contours do not include other tissues than the tissue of interest (GM; magenta, WM; blue). 
-    - [ ] If you found any of the issues above, tweak the segmentation and re-run *fMRIPrep*.
+    - [ ] If you found any of the issues above, do not use the CompCor confounds for nuisance regression.
```
```suggestion
    - [ ] **Do not** use the CompCor confounds for nuisance regression if you find any of the issues above.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
-  If observed, tweak the normalization and re-run *fMRIPrep*.
-
-### Surface reconstruction
+  If observed, use the preprocessed BOLD images in subject space for the analysis rather than MNI.
```
What does this mean?
```suggestion
  If observed, use the preprocessed BOLD images in subject space for the analysis rather than MNI.
```

celprov replied:
If we run into coregistration issue, we will perform the analysis in subject space

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
 - [ ] Check for severe stretching or distortion of the T1w.
-  If observed, tweak the normalization and re-run *fMRIPrep*.
-
-### Surface reconstruction
```
Why does surface reconstruction disappear?

celprov replied:
I categorized those criteria as exclusion criteria not as QA criteria, because I did not find a way to modify surface reconstruction if it fails. As such, the scan has no choice but to be excluded.

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
-- [ ] Verify that in the field `Multi-echo EPI sequence` four echoes were detected.
+- [ ] Check that the repetition time is the expected 1.6s. 
+  If not:
+    - [ ] Investigate whether the TR has been mistakenly reported in the Nifti header, fix it and re-run *fMRIPrep*.
```
```suggestion
    - [ ] Investigate whether the TR metadata in the sidecar JSON file is wrong.
      Fix it and re-run *fMRIPrep*.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
+- [ ] Check that the repetition time is the expected 1.6s. 
+  If not:
+    - [ ] Investigate whether the TR has been mistakenly reported in the Nifti header, fix it and re-run *fMRIPrep*.
+    - [ ] If the session has been acquired with the wrong TR, [exclude that session](#textual-summary-1).
```
```suggestion
    - [ ] [Exclude the whole session](#textual-summary-1) if it was acquired with the wrong TR.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
+  If not:
+    - [ ] Investigate whether the TR has been mistakenly reported in the Nifti header, fix it and re-run *fMRIPrep*.
+    - [ ] If the session has been acquired with the wrong TR, [exclude that session](#textual-summary-1).
+- [ ] Verify that slice timing correction was applied. 
```
Not sure we want to do this.

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,55 +1,51 @@
+- [ ] Verify that slice timing correction was applied. 
+  If not, fix the problem and re-run *fMRIPrep*.
+- [ ] Verify that susceptibility distortion correction was applied. 
+  If not, check that fieldmaps were present in the unprocessed data, that they are no typos in their filename and that the field `B0FieldSource` and `B0FieldIdentifier` have been correctly setup in the JSON sidecars.
```
```suggestion
  If not, check that fieldmaps were present in the unprocessed data, that there are no typos in their filename, and that the fields `B0FieldSource` and `B0FieldIdentifier` have been correctly set in the JSON sidecars.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
     - [ ] Verify that the image boundaries as well as the anatomical landmarks, such as the ventricles and the corpus callosum, appear in the same place when toggling between images.
     - [ ] Verify that the white and pial surface outline (red and blue lines) correspond well to the tissues boundaries in the BOLD image.
-- [ ] Check that no residual susceptibility distortion affects the BOLD image. Susceptibility distortion manifests as signal drop-outs or brain distortions.
+    - [ ] If issues are observed in the points above, tweak the co-registration by increasing its degrees of freedom using the `--bold2t1w-dof 9` flag of *fMRIPrep* and re-run *fMRIPrep*.
```
```suggestion
    - [ ] Increase the degrees of freedom of the co-registration transform setting `--bold2t1w-dof 9` to the *fMRIPrep* call, and re-run *fMRIPrep* if issues are observed at any point above.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
     - [ ] Verify that the white and pial surface outline (red and blue lines) correspond well to the tissues boundaries in the BOLD image.
-- [ ] Check that no residual susceptibility distortion affects the BOLD image. Susceptibility distortion manifests as signal drop-outs or brain distortions.
+    - [ ] If issues are observed in the points above, tweak the co-registration by increasing its degrees of freedom using the `--bold2t1w-dof 9` flag of *fMRIPrep* and re-run *fMRIPrep*.
+    - [ ] If that did not solve co-registration problem, assign the fMRIPrep flag `--bold2t1w-init header` and re-run *fMRIPrep*.
```
```suggestion
    - [ ] If that did not solve the co-registration problem, set `--bold2t1w-init header` instead and re-run *fMRIPrep*.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
-- [ ] Check that no residual susceptibility distortion affects the BOLD image. Susceptibility distortion manifests as signal drop-outs or brain distortions.
+    - [ ] If issues are observed in the points above, tweak the co-registration by increasing its degrees of freedom using the `--bold2t1w-dof 9` flag of *fMRIPrep* and re-run *fMRIPrep*.
+    - [ ] If that did not solve co-registration problem, assign the fMRIPrep flag `--bold2t1w-init header` and re-run *fMRIPrep*.
+    - [ ] If co-registration is still faulty, [exclude this session](#alignment-of-functional-and-anatomical-mri-data).
```
```suggestion
    - [ ] [Exclude this session](#alignment-of-functional-and-anatomical-mri-data) if the co-registration performance remains insufficient.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
+    - [ ] If that did not solve co-registration problem, assign the fMRIPrep flag `--bold2t1w-init header` and re-run *fMRIPrep*.
+    - [ ] If co-registration is still faulty, [exclude this session](#alignment-of-functional-and-anatomical-mri-data).
+
+- [ ] Check that no large residual susceptibility distortion affects the BOLD image. Susceptibility distortion manifests as signal drop-outs or brain distortions.
```
```suggestion
- [ ] Check that no large residual susceptibility distortion affects the BOLD image.
  Susceptibility distortion manifests as signal drop-outs or brain distortions.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
+    - [ ] If co-registration is still faulty, [exclude this session](#alignment-of-functional-and-anatomical-mri-data).
+
+- [ ] Check that no large residual susceptibility distortion affects the BOLD image. Susceptibility distortion manifests as signal drop-outs or brain distortions.
+    - [ ] If large susceptibility distortion remains, double-check that [susceptibility distortion correction was applied correctly by *fMRIPrep*](#textual-summary).
```
Not sure what this means.
```suggestion
<!--
    - [ ] Check the [susceptibility distortion estimation](#textual-summary) if large susceptibility distortion remains.-->
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -60,14 +56,23 @@
     Note that a hole in the middle of the brain mask is not a problem, as this will not disrupt co-registration.
     Note also that it is not a problem that the brain mask is loose around the brain.
-- [ ] Verify that the temporal CompCor mask appears as a collection of points scattered around the brain (with typically higher density in the outer rim of the frontal lobe and close to the brainstem) and does not form suspiciously aggregated shapes (an indicator of an underlying artifact).
+    If the brain mask does not appear as it should, make sure you should not 
```
I think Yoda wrote this, and I can't be sure where this was going.

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+### Surface reconstruction
+
+!!! note "If you used the `--fs-no-reconall` flag to skip surface-based preprocessing, the section of the report will not exist"
+
```
We are using it, so no reason to have this.
```suggestion
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+- [ ] Verify that the WM and pial surface do not cross or overlap each other.
+- [ ] Verify that the pial surface (red line) does not extend past the actual pial boundary.
+
+!!! Tip "Evaluating the quality of brain surfaces"
```
```suggestion
!!! tip "Evaluating the quality of brain surfaces"
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+!!! Tip "Evaluating the quality of brain surfaces"
+
+    As we will proceed with voxel-wise analysis, re-running *fMRIPrep* is necessary only when the reconstructed surfaces are extremely inaccurate.
```
I think the voxel-wise analysis leaked from the frontiers sops.
```suggestion
    As we will proceed with voxel-wise analysis, re-running *fMRIPrep* is necessary only when the reconstructed surfaces are extremely inaccurate.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+# Exclusion criteria for preprocessed task fMRI
+
+!!! important "We do not put together exclusion criteria for preprocessed task fMRI"
+
+    Our QC protocol is very lenient for task fMRI as those scans have been primarily acquired for QC purposes or to aid in methodological development, rather than holding scientific significance. 
+    Task fMRI are excluded only in case of extreme distortion of the image, however such cases would have been already captured by [the first QC checkpoint](../data-management/mriqc.md#visualizing-mriqcs-individual-reports).
```
I need more time to think about this.

celprov replied:
After reflection, they are almost all quality criteria (typically the presence of a coil failure or the fact that the TR is different from other sessions) apply to QCT and BHT as well IMHO.
I'm rewriting it in the next commit

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+- [ ] If the RSfMRI has been acquired with a TR different from 1.6s, exclude that scan. It is likely the other fMRI scans have been mistakenly acquired with the  
+  wrong TR, so double-check the task fMRI.
+
```
```suggestion
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+### Textual summary
+
+- [ ] If the RSfMRI has been acquired with a TR different from 1.6s, exclude that scan. It is likely the other fMRI scans have been mistakenly acquired with the  
+  wrong TR, so double-check the task fMRI.
```
line breaks. Also, it feels like there's a lot of overlap with something I read above.
```suggestion
- [ ] If the RSfMRI has been acquired with a TR different from 1.6s, exclude that scan.
  It is likely the other fMRI scans have been mistakenly acquired with the wrong TR, so double-check the task fMRI.
```

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -76,8 +81,65 @@ The carpet plot MUST appear homogeneous:
+
+### BOLD summary
+
+- [ ] Check for extensive signal drops that may reveal a coil failure. If such pattern is observed, exclude the RSfMRI scan.
```
```suggestion
- [ ] Check for outstanding signal drops that may reveal a coil failure.
  If such a pattern is observed, exclude the RSfMRI scan.
```

oesteban reviewed (commented):
getting there :)

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>

celprov committed:
fix: remove the statement that we do not define exclusion criteria for
QCT and BHT.
fix: precise that some carpetplot QA criteria are only for RSfMRI
fix: apply code suggestions from review
fix: make sure all exclusion criteria are pointed by some text either in
preprocessing.md or in the QA criteria

celprov committed:
Apply suggestions from code review

oesteban commented:
I'll check on this one tomorrow :)

oesteban committed:
fix: undo spurious changes to processing.md

oesteban committed:
sty: minimal edits

oesteban committed:
enh: reorganize and add an intro

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,83 +1,124 @@
-- [ ] Verify that the pial surface (red line) does not extend past the actual pial boundary.
 
-!!! Tip "Evaluating the quality of brain surfaces"
+  !!! note "We can afford to work in subject space because our study is based on a single subject"
 
```
Let's remove this comment:

```suggestion
```

(i) This is not really a QA/QC criteria, but rather a study design consideration regarding the analysis.
(ii) Being a design consideration, the language of "we can afford" is inappropriate. Design decisions are based on what you want and need to do to achieve something the best way possible. Informally, this can be okay, but writing it in the protocol is a bit misfortunate.

oesteban commented on docs/processing/qaqc-criteria-preprocessed.md:
```diff
@@ -1,83 +1,124 @@
+    - [ ] Increase the degrees of freedom of the co-registration transform setting `--bold2t1w-dof 9` to the *fMRIPrep* call, and re-run *fMRIPrep* if issues are observed at any point above.
+    - [ ] If that did not solve the co-registration problem, set `--bold2t1w-init header` instead and re-run *fMRIPrep*.
+    - [ ] [Exclude this session](#alignment-of-functional-and-anatomical-mri-data) if the co-registration performance remains insufficient.
+<!--
```
```suggestion
<!-- https://github.com/TheAxonLab/hcph-sops/issues/453
```

oesteban reviewed (approved):
LGTM

I've done a bit of reorganization and added a QA/QC section that was missing IMHO. That will help avoid repetitions and contextualize some stuff that was coming out a bit out of the blue.

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>
